File: Algorithms/JAVA/Sorting_Algorithms/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {
    public static void main(String[] args) {
        int[] arr = {4,5,1,2,3};
        insertion(arr);
        System.out.println(Arrays.toString(arr));
    }

    private static void insertion(int[] arr) {
        for (int i = 0; i < arr.length-1; i++) {
            for (int j = i+1; j > 0; j--) {
                if(arr[j]<arr[j-1])
                    swap(arr,j,j-1);
                else
                    break;
            }
        }
    }

    private static void swap(int[] arr, int first, int second) {
        int temp = arr[first];
        arr[first]=arr[second];
        arr[second]=temp;
    }
}

File: Algorithms/Python/Sorting_Algorithms/INSERTION_SORT.py
def insertionSort(arr):

	for i in range(1, len(arr)):

		k = arr[i]
		j = i-1
		while j >=0 and k < arr[j] :
				arr[j+1] = arr[j]
				j -= 1
		arr[j+1] = k

arr = [18, 12, 14, 15, 6]
insertionSort(arr)
print("Sorted array is : ",arr)

